user_tools/src/spark_rapids_tools/tools/top_candidates.py

```diff
@@ -59,22 +59,29 @@ def get_filtered_apps_count(self) -> int:
 
     def _filter_apps(self) -> None:
         """
-        Filters the applications based on the eligible categories (Small/Medium/Large)
+        Filters the applications based on the eligible categories (Small/Medium/Large) and total core seconds threshold.
         """
         if not self._has_tools_processed_apps():
             self.filtered_apps = pd.DataFrame(columns=self.tools_processed_apps.columns)
             return
 
         category_col_name = self.props.get('categoryColumnName')
         eligible_categories = self.props.get('eligibleCategories')
+        # Filter based on eligible categories (Small/Medium/Large)
         filter_condition = self.tools_processed_apps[category_col_name].isin(eligible_categories)
-        self.filtered_apps = self.tools_processed_apps[filter_condition]
+        # Filter based on total core seconds threshold
+        total_core_sec_col = self.props.get('totalCoreSecCol')
+        total_core_sec_threshold = self.props.get('totalCoreSecThreshold')
+        total_core_sec_cond = self.tools_processed_apps[total_core_sec_col] > total_core_sec_threshold
+        self.filtered_apps = self.tools_processed_apps[filter_condition][total_core_sec_cond]
```

Comment:
Can we create a single filter condition by combining individual conditions? This will avoid multiple passes over the DF for each condition.

We could follow a similar approach as in the heuristics case: start with a base condition and then incrementally add each condition. It could also help in extensibility.

Reply:
thanks @parthosa, updated this.